File: GymApp/composeApp/src/commonMain/kotlin/com/gym/gymapp/ui/components/GymBottomBar.kt
package com.gym.gymapp.ui.components

import androidx.compose.animation.*
import androidx.compose.animation.core.Spring
import androidx.compose.animation.core.spring
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.interaction.MutableInteractionSource
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.Icon
import androidx.compose.material.Text
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Home
import androidx.compose.material.icons.filled.DateRange
import androidx.compose.material.icons.filled.Person
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.material.icons.automirrored.filled.List
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.material3.MaterialTheme

import androidx.compose.material.icons.filled.Payments
import androidx.compose.material.icons.filled.Assignment
import androidx.compose.material.icons.filled.Campaign

enum class MainTab(val icon: ImageVector, val label: String) {
    HOME(Icons.Default.Home, "Home"),
    MEMBERS(Icons.AutoMirrored.Filled.List, "Members"),
    PAYMENTS(Icons.Default.Payments, "Pay"),
    PLANS(Icons.Default.Assignment, "Plans"),
    PROFILE(Icons.Default.Person, "Profile")
}

@Composable
fun GymBottomBar(
    selectedTab: MainTab,
    onTabSelected: (MainTab) -> Unit
) {
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .padding(bottom = 20.dp, start = 16.dp, end = 16.dp)
            .height(72.dp), // Increased height
        contentAlignment = Alignment.Center
    ) {
        Row(
            modifier = Modifier
                .background(MaterialTheme.colorScheme.surface, RoundedCornerShape(36.dp))
                .padding(horizontal = 8.dp, vertical = 6.dp)
                .wrapContentWidth(),
            horizontalArrangement = Arrangement.spacedBy(8.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {

            MainTab.entries.forEach { tab ->
                val isSelected = selectedTab == tab
                
                BottomBarItem(
                    tab = tab,
                    isSelected = isSelected,
                    onClick = { onTabSelected(tab) }
                )
            }
        }
    }
}

@OptIn(ExperimentalAnimationApi::class)
@Composable
private fun BottomBarItem(
    tab: MainTab,
    isSelected: Boolean,
    onClick: () -> Unit
) {
    val interactionSource = remember { MutableInteractionSource() }
    
    Box(
        modifier = Modifier
            .padding(1.dp) // Reduced padding between items
            .clip(RoundedCornerShape(24.dp))
            .background(if (isSelected) MaterialTheme.colorScheme.onSurface.copy(alpha = 0.05f) else Color.Transparent)
            .clickable(
                interactionSource = interactionSource,
                indication = null,
                onClick = onClick
            )
            .padding(horizontal = if (isSelected) 8.dp else 6.dp, vertical = 6.dp) // Snappier horizontal padding
            .animateContentSize(
                animationSpec = spring(
                    dampingRatio = Spring.DampingRatioLowBouncy,
                    stiffness = Spring.StiffnessMediumLow
                )
            ),
        contentAlignment = Alignment.Center
    ) {
        Row(
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.Center
        ) {
            Box(
                modifier = Modifier
                    .size(36.dp) // Larger icons container
                    .clip(CircleShape)
                    .background(if (isSelected) MaterialTheme.colorScheme.primary else Color.Transparent),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    imageVector = tab.icon,
                    contentDescription = tab.label,
                    tint = if (isSelected) MaterialTheme.colorScheme.onPrimary else MaterialTheme.colorScheme.onSurface.copy(alpha = 0.5f),
                    modifier = Modifier.size(24.dp) // Larger icons
                )
            }

            AnimatedVisibility(
                visible = isSelected,
                enter = fadeIn(animationSpec = spring(stiffness = Spring.StiffnessMediumLow)) + 
                        expandHorizontally(animationSpec = spring(stiffness = Spring.StiffnessMediumLow)),
                exit = fadeOut(animationSpec = spring(stiffness = Spring.StiffnessMediumLow)) + 
                        shrinkHorizontally(animationSpec = spring(stiffness = Spring.StiffnessMediumLow))
            ) {
                Row {
                    Spacer(modifier = Modifier.width(6.dp))
                    Text(
                        text = tab.label,
                        color = MaterialTheme.colorScheme.onSurface,
                        fontSize = 11.sp, // Slightly larger font
                        fontWeight = FontWeight.Bold,
                        maxLines = 1,
                        overflow = TextOverflow.Ellipsis
                    )
                    Spacer(modifier = Modifier.width(4.dp))
                }
            }
        }

    }
}

File: gym_flutter_app/lib/ui/widgets/gym_bottom_bar.dart
import 'package:flutter/material.dart';

enum MainTab { home, members, payments, plans, profile }

class GymBottomBar extends StatelessWidget {
  final MainTab selectedTab;
  final Function(MainTab) onTabSelected;

  const GymBottomBar({
    super.key,
    required this.selectedTab,
    required this.onTabSelected,
  });

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    
    return Container(
      padding: const EdgeInsets.only(bottom: 20, left: 16, right: 16),
      height: 92,
      child: Center(
        child: Container(
          padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 6),
          decoration: BoxDecoration(
            color: theme.colorScheme.surface,
            borderRadius: BorderRadius.circular(36),
          ),
          child: Row(
            mainAxisSize: MainAxisSize.min,
            children: MainTab.values.map((tab) {
              final isSelected = selectedTab == tab;
              return _BottomBarItem(
                tab: tab,
                isSelected: isSelected,
                onTap: () => onTabSelected(tab),
              );
            }).toList(),
          ),
        ),
      ),
    );
  }
}

class _BottomBarItem extends StatelessWidget {
  final MainTab tab;
  final bool isSelected;
  final VoidCallback onTap;

  const _BottomBarItem({
    required this.tab,
    required this.isSelected,
    required this.onTap,
  });

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);

    return GestureDetector(
      onTap: onTap,
      child: AnimatedContainer(
        duration: const Duration(milliseconds: 300),
        curve: Curves.easeInOut,
        padding: EdgeInsets.symmetric(horizontal: isSelected ? 10 : 6, vertical: 6),
        margin: const EdgeInsets.symmetric(horizontal: 2),
        decoration: BoxDecoration(
          color: isSelected ? theme.colorScheme.onSurface.withValues(alpha: 0.05) : Colors.transparent,
          borderRadius: BorderRadius.circular(24),
        ),
        child: Row(
          children: [
            Container(
              width: 36,
              height: 36,
              decoration: BoxDecoration(
                color: isSelected ? theme.colorScheme.primary : Colors.transparent,
                shape: BoxShape.circle,
              ),
              child: Icon(
                _getIcon(tab),
                color: isSelected ? Colors.black : theme.colorScheme.onSurface.withValues(alpha: 0.5),
                size: 24,
              ),
            ),
            if (isSelected) ...[
              const SizedBox(width: 6),
              Text(
                _getLabel(tab),
                style: theme.textTheme.bodySmall?.copyWith(
                  color: theme.colorScheme.onSurface,
                  fontWeight: FontWeight.bold,
                  fontSize: 11,
                ),
              ),
              const SizedBox(width: 4),
            ],
          ],
        ),
      ),
    );
  }

  IconData _getIcon(MainTab tab) {
    switch (tab) {
      case MainTab.home: return Icons.home;
      case MainTab.members: return Icons.format_list_bulleted;
      case MainTab.payments: return Icons.payments;
      case MainTab.plans: return Icons.assignment;
      case MainTab.profile: return Icons.person;
    }
  }

  String _getLabel(MainTab tab) {
    switch (tab) {
      case MainTab.home: return "Home";
      case MainTab.members: return "Members";
      case MainTab.payments: return "Pay";
      case MainTab.plans: return "Plans";
      case MainTab.profile: return "Profile";
    }
  }
}
